File: backend/app/schemas/payment.py
"""
Pydantic schemas for payment and subscription
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class CheckoutSessionCreate(BaseModel):
    """创建结账会话的请求。旧字段名保留给当前前端兼容。"""
    plan: str = Field(..., description="订阅计划: pro_monthly/pro_yearly，兼容 monthly/yearly")
    success_url: str = Field(..., description="支付成功后的回调URL")
    cancel_url: str = Field(..., description="取消支付后的回调URL")
    provider: str = Field(default="stripe", description="支付通道")


class CheckoutSessionResponse(BaseModel):
    """结账会话响应。兼容旧 checkout_url/session_id，同时暴露新订单字段。"""
    checkout_url: str = Field(..., description="托管结账页面URL")
    session_id: str = Field(..., description="兼容旧前端的订单ID")
    provider: str = Field(default="stripe")
    order_id: str
    merchant_order_id: str
    qr_code_url: Optional[str] = None
    expires_at: Optional[datetime] = None


class PaymentCaptureResponse(BaseModel):
    provider: str
    order_id: str
    merchant_order_id: str
    status: str
    current_period_end: Optional[datetime] = None


class PaymentProviderOption(BaseModel):
    key: str
    enabled: bool
    display_name: str
    settlement: str
    supports_subscription: bool
    supports_one_time: bool


class PaymentProviderList(BaseModel):
    providers: list[PaymentProviderOption]


class SubscriptionResponse(BaseModel):
    """订阅信息响应"""
    has_subscription: bool
    status: Optional[str] = None  # active, past_due, canceled, etc.
    plan: Optional[str] = None  # price_id
    current_period_end: Optional[datetime] = None
    cancel_at_period_end: bool = False


class WebhookEvent(BaseModel):
    """Stripe Webhook事件"""
    type: str
    data: dict

File: src/services/api/payment.ts
import apiClient from '../http'

export interface CheckoutSessionRequest {
  plan: 'monthly' | 'yearly'
  success_url: string
  cancel_url: string
  provider?: PaymentProviderKey
}

export interface CheckoutSessionResponse {
  checkout_url: string
  session_id: string
  provider: PaymentProviderKey
  order_id: string
  merchant_order_id: string
  qr_code_url?: string | null
  expires_at?: string | null
}

export type PaymentProviderKey =
  | 'stripe'
  | 'paypal'
  | 'epay'
  | 'alipay'
  | 'wechat'
  | 'tokenpay'
  | 'bepusdt'
  | 'epusdt'
  | 'okpay'
  | 'gmpay'

export interface PaymentProviderOption {
  key: PaymentProviderKey
  enabled: boolean
  display_name: string
  settlement: string
  supports_subscription: boolean
  supports_one_time: boolean
}

export interface PaymentProviderList {
  providers: PaymentProviderOption[]
}

export interface PaymentCaptureResponse {
  provider: PaymentProviderKey
  order_id: string
  merchant_order_id: string
  status: string
  current_period_end?: string | null
}

export interface SubscriptionInfo {
  has_subscription: boolean
  status?: string
  plan?: string
  current_period_end?: string
  cancel_at_period_end: boolean
}

export const paymentAPI = {
  async createCheckoutSession(
    data: CheckoutSessionRequest,
  ): Promise<CheckoutSessionResponse> {
    const response = await apiClient.post<CheckoutSessionResponse>(
      '/api/v1/payment/create-checkout-session',
      data,
    )
    return response.data
  },

  async listProviders(): Promise<PaymentProviderList> {
    const response = await apiClient.get<PaymentProviderList>('/api/v1/payment/providers')
    return response.data
  },

  async captureOrder(merchantOrderId: string): Promise<PaymentCaptureResponse> {
    const response = await apiClient.post<PaymentCaptureResponse>(
      `/api/v1/payment/orders/${merchantOrderId}/capture`,
    )
    return response.data
  },

  async getSubscription(): Promise<SubscriptionInfo> {
    const response = await apiClient.get<SubscriptionInfo>('/api/v1/payment/subscription')
    return response.data
  },

  async cancelSubscription(): Promise<{ message: string; cancel_at: string }> {
    const response = await apiClient.post('/api/v1/payment/cancel-subscription')
    return response.data
  },

  async reactivateSubscription(): Promise<{ message: string }> {
    const response = await apiClient.post('/api/v1/payment/reactivate-subscription')
    return response.data
  },
}
